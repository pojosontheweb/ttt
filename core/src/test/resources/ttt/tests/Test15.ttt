<%@ page import="com.xyz.myapp.Foo" %>
<%@ page contentType="text/html" %>
<%--
    JSP comment can span
    appear anywhere
    and are ignored when template gets compiled
--%>
<%@ page extends="java.io.Serializable" %>
<%!
    Foo foo;
    int bar;
%>
<div class="foo">
    <span><%= foo.getBar() %></span>
    <ul>
    <% for(int i = 0; i < bar; i++) { %>
        <li>i equals <%= i %> </li>
    <% } %>
    </ul>
</div>