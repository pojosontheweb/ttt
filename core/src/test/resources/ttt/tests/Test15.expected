package ttt.tests;

import com.xyz.myapp.Foo;

public class Test15 extends com.pojosontheweb.ttt.Template implements java.io.Serializable {

	private final Foo foo;
	private final int bar;

	/**
	 * Creates an instance of this template.
	 *
	 */
	public Test15(Foo foo, int bar) {
		this.foo = foo;
		this.bar = bar;
	}

	@Override
	public String getContentType() {
		return "text/html";
	}

	@Override
	public void doRender(com.pojosontheweb.ttt.TttWriter out) throws Exception {
		write(out, "<div class=\"foo\">\n    <span>" );
		write(out,  foo.getBar()  );
		write(out, "</span>\n    <ul>\n    " );
		for(int i = 0; i < bar; i++) { 
		write(out, "\n        <li>i equals " );
		write(out,  i  );
		write(out, " </li>\n    " );
		} 
		write(out, "\n    </ul>\n</div>" );
	}
}
